src/entry.c

```diff
@@ -1483,3 +1484,113 @@ void
 unlock_entry( const struct stumpless_entry *entry ) {
   config_unlock_mutex( entry->mutex );
 }
+
+
+/*
+ * formating entry_to_string values
+ */
+static size_t append_key_value_pair(char *dest, size_t offset, const char *key, const char *value, size_t value_length) {
+    size_t key_len = strlen(key);
```

Comment:
It would be safer to pass in a key size parameter, the same as you've done with the value parameter, and avoid this call to `strlen`.